src/cpp/server/include/server/ServerOptions.gen.hpp

```diff
@@ -115,8 +115,8 @@ class GeneratedOptions
       value<bool>(&wwwUseEmulatedStack_)->default_value(false),
       "Indicates whether or not to use GWT's emulated stack.")
       ("www-thread-pool-size",
-      value<int>(&wwwThreadPoolSize_)->default_value(2),
-      "The size of the threadpool from which requests will be serviced. This may be increased to enable more concurrency, but should only be done if the underlying hardware has more than 2 cores. It is recommended to use a value that is <= to the number of hardware cores, or <= to two times the number of hardware cores if the hardware utilizes hyperthreading.")
+      value<int>(&wwwThreadPoolSize_)->default_value(6),
```

Comment:
What do you think about setting a default value of `-1` (or some similar proxy value), and then computing an appropriate number of threads at runtime, based on the number of available CPUs? (Maybe using half of the available cores, with a minimum of 2 and maximum of 8?)

Reply:
Good question, though I think even if you have 1 core, you might need more threads to avoid bottlenecks. Ideally we would not have any requests that block on the request thread and then it would totally make sense to do that. But since like the login request can block a thread for 5s, it's pretty easy to bottleneck even on a small site. One thing that might make sense is to keep the default of 2 for RStudio Server and bump it only for workbench? That would make sense if most RStudio Server users were only one or two users at a time. Giving up a little bit of runtime efficiency seems like an ok tradeoff to avoid major disruption though as seen by some of the larger workbench deployments.

Reply:
That sounds good to me.